### PR TITLE
Emma Andrews tickets

### PR DESCRIPTION
**Pull request by ELAndrews**
## What does this PR do?

The implementation routers, models and middleware for tickets and comments
 
## Description of tasks completed

- tickets : GetAll(), getById(), getByCategory(), postNew()
- comments: getAll(), getById(), postNew()
- middleware: validateUsername(), validateRequestbody x 3(comment, ticket, user)
 
 # How Has This Been Tested?

Please describe steps taken to ensure this feature is functional and does not break other features:

Postman has been used to check calls as well as HTTPie

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, if relevant
- [ ] I have run the app using my feature and ensured that no functionality is broken
- [ ] My changes generate no new warnings
- [ ] There are no merge conflicts

### CODE REVIEW

File: src/middleware/error-handling-middleware.js [hunk 1]
```diff
@@ -2,7 +2,82 @@ const Users = require("../models/user-models");
...
+    req.body.status ||
+    req.body.category_id ||
+    req.body.student_id
+  ) {
+    res
+      .status(400)
+      .json(
+        `Please provide a valide title, content, status, category and student`
+      );
+  } else {
```
Comment (nedssoft) [inline src/middleware/error-handling-middleware.js]: I think you meant `!req.body.title` ?

File: src/middleware/error-handling-middleware.js [hunk 2]
```diff
@@ -2,7 +2,82 @@ const Users = require("../models/user-models");
...
+    req.body.password &&
+    req.body.role
+  ) {
+    next();
+  } else if (
+    req.body.first_name ||
+    req.body.last_name ||
+    req.body.username ||
+    req.body.password ||
+    req.body.role
```
Comment (nedssoft) [inline src/middleware/error-handling-middleware.js]: Did you mean if not those attributes?

File: src/routers/comment-routers.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      res.status(500).json(error.message);
+    });
+});
+
+// ---------------- POST ---------------- //
+
+router.post("/", (req, res) => {
+  Comments.addNewComment(req.body)
+    .then(id => {
+      console.log(id);
```
Comment (nedssoft) [inline src/routers/comment-routers.js]: console.log?

File: src/routers/ticket-routers.js [hunk 4]
```diff
@@ -1,4 +1,49 @@
...
+    })
+    .catch(error => {
+      res.status(500).json(error.message);
+    });
+});
+
+router.get("/category/:id", (req, res) => {
+  Tickets.getTicketByCategory(req.params.id)
+    .then(tickets => {
+      console.log(tickets);
```
Comment (nedssoft) [inline src/routers/ticket-routers.js]: This is not needed in the PR, is it?

File: src/routers/ticket-routers.js [hunk 5]
```diff
@@ -1,4 +1,49 @@
...
+      res.status(500).json(error.message);
+    });
+});
+
+// ---------------- POST ---------------- //
+
+router.post("/", (req, res) => {
+  Tickets.addNewTicket(req.body)
+    .then(id => {
+      console.log(id);
```
Comment (nedssoft) [inline src/routers/ticket-routers.js]: ?

File: src/routers/comment-routers.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+const router = require("express").Router();
+const Comments = require("../database/models/comment-models");
```
Comment (nedssoft) [inline src/routers/comment-routers.js]: Model name should be singular, e.g `Comment`

**Review by nedssoft — COMMENTED** [review]
Good job;  I made a few recommendations that you might want to take into considerations

File: src/middleware/error-handling-middleware.js (hunk 1 above)
Comment (ELAndrews) [reply src/middleware/error-handling-middleware.js]: well spotted. Have amended

File: src/middleware/error-handling-middleware.js (hunk 2 above)
Comment (ELAndrews) [reply src/middleware/error-handling-middleware.js]: Same issue as before. I have also amended this as well

File: src/routers/comment-routers.js (hunk 6 above)
Comment (ELAndrews) [reply src/routers/comment-routers.js]: Changed to singular

File: src/routers/ticket-routers.js (hunk 5 above)
Comment (ELAndrews) [reply src/routers/ticket-routers.js]: So I clearly forgot to complete this function. I have no completed it.

File: src/routers/ticket-routers.js (hunk 4 above)
Comment (ELAndrews) [reply src/routers/ticket-routers.js]: very true ...

**Review by niklasbec — APPROVED** [review]

**Pull request merged**